peachjam/views/documents.py

```diff
@@ -229,6 +229,24 @@ def render_to_response(self, context, **response_kwargs):
         return response
 
 
+class DocumentAttachmentView(DocumentDetailView):
+    def render_to_response(self, context, **response_kwargs):
+        file = self.object.attachedfiles_set.filter(
+            filename=self.kwargs["filename"]
+        ).first()
+        if file and not file.private:
+            if getattr(file.file.storage, "custom_domain", None):
+                # use the storage's custom domain to serve the file
+                return redirect(file.file.url)
+
+            file_bytes = file.file.open().read()
+            response = HttpResponse(file_bytes, content_type=file.mimetype)
+            response["Content-Disposition"] = f"attachment; filename={file.filename}"
```

Comment:
I have learnt the hard way that the filenames can be horribly dirty. Clean it up before sending it:

```python
filename = re.sub(r'[^A-Za-z0-9._-]', '', file.filename)
```